contributors/design-proposals/daemonset-update.md

```diff
@@ -165,22 +187,38 @@ DaemonSets on the apiserver.
 
 For each pending DaemonSet updates, it will:
 
-1. Find all pods whose labels are matched by DaemonSet `.spec.selector`. 
-   - If `OwnerReference` is implemented for DaemonSets, filter out pods that
-     aren't controlled by this DaemonSet too.
+1. Find all pods controlled by DaemonSet
 1. Check `DaemonSetUpdateStrategy`:
    - If `OnDelete`: do nothing
    - If `RollingUpdate`:
-     - Find all daemon pods that belong to this DaemonSet using label selectors.
-       - Pods with "pod-template-generation" value equal to the DaemonSet's
-         `.spec.templateGeneration` are new pods, otherwise they're old pods.
-       - Note that pods without "pod-template-generation" labels (e.g. DaemonSet
-         pods created before RollingUpdate strategy is implemented) will be
-         seen as old pods.
+     - Find existing PodTemplates owned by DaemonSet, and sort them by the value
+       of `pod-template-generation` label
+     - Clean up PodTemplates based on `.spec.revisionHistoryLimit`
```

Comment:
> In your example, there will be 3 more PodTemplates created for OnDelete. Is that expected? In this case, it seems that a single history with all three changes make more sense (since you want to trigger all 3 changes at once).

No new PodTemplate should be generated in the OnDelete case. At least, when a Deployment is paused, we won't create a new ReplicaSet for it.

> @kargakis another question, do people generally use pause before triggering a rollout (your example), or in the middle of it?

Based on feedback from others and on my own usage, most common is the flow I explained above: use `pause` before triggering. Ideally, for manual deployments (eg. prod rollouts), I think we want to run one-off rollouts and always run with `paused=true`. Pausing in the middle or at any point in the rollout is auto-pause (https://github.com/kubernetes/kubernetes/issues/11505) and that ability solves many other use cases (canaries, A/B, hooks).

> Will this usage be different in Deployments vs. DaemonSet?

IMO, the flow described above applies equally to DaemonSets and StatefulSets. Auto-pause will probably differ slightly.